metadata/ota_metadata/metadata_gen.py

```diff
@@ -216,6 +232,72 @@ def gen_metadata(
         if f.is_file() and not f.is_symlink():
             regulars.append(str(f.relative_to(target_dir)))
 
+    # symlinks.txt
+    # format:
+    # mode,uid,gid,'path/to/link','path/to/target'
+    # ex: 0777,1000,1000,'path/to/link','path/to/target'
+    # NOTE: mode is always 0777.
+    symlink_list = []
+    target_regular_set = set()
+    target_dir_set = set()
+
+    for d in symlinks:
+
+        symlink_target_path = os.readlink(os.path.join(target_dir, d))
+        symlink_entry = (
+            f"{_join_mode_uid_gid(target_dir, d)},"
+            f"{_encapsulate(d, prefix=prefix)},"
+            f"{_encapsulate(symlink_target_path)}"
+        )
+        symlink_list.append(symlink_entry)
+
+        # Do nothing when igore pattern 
+        # does not have "home/autoware...../build" or "home/autoware...../src" definition
+        if check_symlink is False:
+            continue
+
+        # Do nothing if the targeted file link 
+        # is not under "home/autoware...../build" or "home/autoware...../src"
+        should_check = False
+        for pattern in check_patterns:
+            if bool(re.search(pattern, str(symlink_target_path))) is True:
+                should_check = True
+                break
+        if should_check is False:
+            continue
```

Comment:
(nit) this might be difficult to read.
```suggestion
if not any(re.search(pattern, str(symlink_target_path)) for pattern in check_patterns):
    continue
```